api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -705,9 +729,71 @@ async function getExams(
         retakeTimeInMS: exam.config.retakeTimeInMS,
         passingPercent: exam.config.passingPercent
       },
-      canTake: isExamPrerequisitesMet
+      canTake: false
     };
-  });
+
+    const isExamPrerequisitesMet = checkPrerequisites(user, exam.prerequisites);
+
+    if (!isExamPrerequisitesMet) {
+      availableExam.canTake = false;
+      availableExams.push(availableExam);
+      continue;
+    }
+    // Latest attempt must be:
+    // a) Moderated
+    // b) Past exam config retake time
+    const lastAttempt = attempts.length
+      ? attempts.reduce((latest, current) =>
+          latest.startTimeInMS > current.startTimeInMS ? latest : current
+        )
+      : null;
+
+    if (!lastAttempt) {
+      availableExam.canTake = true;
+      availableExams.push(availableExam);
+      continue;
+    }
+
+    const isRetakeTimePassed =
+      Date.now() >
+      lastAttempt.startTimeInMS +
+        exam.config.totalTimeInMS +
+        exam.config.retakeTimeInMS;
+
+    if (!isRetakeTimePassed) {
+      availableExam.canTake = false;
+      availableExams.push(availableExam);
+      continue;
+    }
+
+    const maybeModerations = await mapErr(
+      this.prisma.examEnvironmentExamModeration.findMany({
+        where: {
+          examAttemptId: lastAttempt.id,
+          NOT: { status: ExamEnvironmentExamModerationStatus.Pending }
+        }
+      })
+    );
+
+    if (maybeModerations.hasError) {
+      logger.error(maybeModerations.error);
+      this.Sentry.captureException(maybeModerations.error);
+      void reply.code(500);
+      return reply.send(
+        ERRORS.FCC_ERR_EXAM_ENVIRONMENT(JSON.stringify(maybeModerations.error))
+      );
```

Comment:
I think you can do something like this:

```suggestion
      return reply.code(500).send(
        ERRORS.FCC_ERR_EXAM_ENVIRONMENT(JSON.stringify(maybeModerations.error))
      );
```